### ISSUE TITLE
Too many messages over network

### ISSUE THREAD

**Issue by ilblackdragon** [nearprotocol/nearcore#1519]
Right now when we start a node - it actually bans all of it's peers because of abuse:
```
[2019-10-22T06:22:53Z WARN  near_network::peer_manager] Banning peer ed25519:2vSf72YzGcWtukicAbhgYvnBfNKoMyHXo7TLLm6LpeEY for abuse (54277 sent, 41361 recv)
[2019-10-22T06:22:57Z INFO  near_client::info] #   20841 H4HP7nw13fJ9yToZa6f5Smey3mnSaTN8aC3bVmE6bTTE V/1  2/1/40 peers ⬇ 128.7kiB/s ⬆ 189.3kiB/s 0.79 bps 0 gas/s CPU: 79%, Mem: 168.4 MiB
[2019-10-22T06:22:58Z WARN  near_network::peer_manager] Banning peer ed25519:EY9mX5FYyR1sqrGwkqCbUrmjgAtXs4DeNaf1sjG9MrkY for abuse (74203 sent, 49076 recv)
[2019-10-22T06:22:58Z WARN  near_network::peer_manager] Banning peer ed25519:8ACS9kqYM3mDvodF4XC2aLsqnAGM3Wkmnr89RAtEntpr for abuse (70492 sent, 66194 recv)
```

Obviously 70k messages per minute is too much.

Also we need to balance a bit more, but with 5 peers uploading 1.6MB seems a lot: `5/1/40 peers ⬇ 29.7kiB/s ⬆ 1.6MiB/s`

**Comment by mfornet** [comment]
I run a node for 20 minutes and collect count and size of messages regarding type of message and sender. Here are the results. Size is in bytes:

```
Count
{'Block': 26192,
 'BlockHeaderRequest': 117,
 'BlockHeaders': 42,
 'BlockRequest': 29753,
 'ChunkOnePart': 96817,
 'Handshake': 9,
 'LastEdge': 4,
 'PeersResponse': 6,
 'Sync': 1060}
Size
{'Block': 80828512,
 'BlockHeaderRequest': 56137,
 'BlockHeaders': 194532,
 'BlockRequest': 981849,
 'ChunkOnePart': 42115395,
 'Handshake': 1530,
 'LastEdge': 824,
 'PeersResponse': 4314,
 'Sync': 424008}

Count
{'ed25519:3n74idReZqNgfr9gHCAYsPeJdPt7dcRQKw3Aa1nacbp6': 301,
 'ed25519:6DY4kDJWZqM88ZwwmDb69yZ2GwyrHFzyJZp5zMeoGKw2': 339,
 'ed25519:8K7NG5v2yvSq4A1wQuqSNvyY334BVq3ohvdu9wgpgjLG': 311,
 'ed25519:AJLCcX4Uymeq5ssavjUCyEA8SV6Y365Mh5h4shqMSTDA': 123193,
 'ed25519:E7mpFQ3GRzFwUmD9Q4MeJz9ui61FBvvGP6cnH4WohySv': 13,
 'ed25519:EY9mX5FYyR1sqrGwkqCbUrmjgAtXs4DeNaf1sjG9MrkY': 4555,
 'ed25519:MJWBf3caWzmGTohREPYCqDL4ZMNpTcxfzP5Wx3pT2i9': 24974,
 'ed25519:We2pjPQqqAt9DKSmwL2vA6L9CAEDgJGFKpavvNwW4sB': 314}
Size
{'ed25519:3n74idReZqNgfr9gHCAYsPeJdPt7dcRQKw3Aa1nacbp6': 60636,
 'ed25519:6DY4kDJWZqM88ZwwmDb69yZ2GwyrHFzyJZp5zMeoGKw2': 67469,
 'ed25519:8K7NG5v2yvSq4A1wQuqSNvyY334BVq3ohvdu9wgpgjLG': 64357,
 'ed25519:AJLCcX4Uymeq5ssavjUCyEA8SV6Y365Mh5h4shqMSTDA': 123191216,
 'ed25519:E7mpFQ3GRzFwUmD9Q4MeJz9ui61FBvvGP6cnH4WohySv': 46006,
 'ed25519:EY9mX5FYyR1sqrGwkqCbUrmjgAtXs4DeNaf1sjG9MrkY': 246452,
 'ed25519:MJWBf3caWzmGTohREPYCqDL4ZMNpTcxfzP5Wx3pT2i9': 866451,
 'ed25519:We2pjPQqqAt9DKSmwL2vA6L9CAEDgJGFKpavvNwW4sB': 64514}```

**Comment by ilblackdragon** [comment]
Wow, 42mb just in ChunkOnePart messages.
But yeah, seems like need to figure out why so many ChunkOnePart messages.

And also block sync is very inefficient right now and probably requesting the same block many times.

**Comment by mfornet** [comment]
In further tests I see the following, out of 39304 block messages received 37614 are repeated. At least 500 blocks are received more than 10 times and one block was received 692 times (this is crazy).

Notice I make no distinction in this tests between messages that are for me, and messages that I should route (anyway those numbers make no sense). I'll explore if network is responsible for such numbers.

**Comment by mfornet** [comment]
@ilblackdragon Blocks and chunks are not being routed right now. It should be solved after: https://github.com/nearprotocol/nearcore/pull/1589. Maybe this is causing troubles in networks with many peers where everyone is not connected to everyone.

**Comment by ilblackdragon** [comment]
@mfornet What's the status with this at the moment?

**Comment by mfornet** [comment]
This issue was fixed, and we keep further tracking on #1586
